Tests/Python/PythonPackagesTests.cs

```diff
@@ -0,0 +1,516 @@
+    model.startprob_ = np.array([0.6, 0.3, 0.1, 0.0])
+    # The transition matrix, note that there are no transitions possible
+    # between component 1 and 3
+    model.transmat_ = np.array([[0.7, 0.2, 0.0, 0.1],
+                               [0.3, 0.5, 0.2, 0.0],
+                               [0.0, 0.3, 0.5, 0.2],
+                               [0.2, 0.0, 0.2, 0.6]])
+    # The means of each component
+    model.means_ = np.array([[0.0,  0.0],
+                             [0.0, 11.0],
+                             [9.0, 10.0],
+                             [11.0, -1.0]])
+    # The covariance of each component
+    model.covars_ = .5 * np.tile(np.identity(2), (4, 1, 1))
+
+    # Generate samples
+    return model.sample(500)"
+            );
+        }
+
+        [Test]
+        public void PomegranateTest()
+        {
+            AssetCode(
+                @"
+from pomegranate import *
+def RunTest():
+    d1 = NormalDistribution(5, 2)
+    d2 = LogNormalDistribution(1, 0.3)
+    d3 = ExponentialDistribution(4)
+    d = IndependentComponentsDistribution([d1, d2, d3])
+    X = [6.2, 0.4, 0.9]
+    return d.log_probability(X)"
+            );
+        }
+
+        [Test]
+        public void LightgbmTest()
+        {
+            AssetCode(
+                @"
+import lightgbm as lgb
+import numpy as np
+import pandas as pd
+from scipy.special import expit
+def RunTest():
+    # Simulate some binary data with a single categorical and
+    # single continuous predictor
+    np.random.seed(0)
+    N = 1000
+    X = pd.DataFrame({
+        'continuous': range(N),
+        'categorical': np.repeat([0, 1, 2, 3, 4], N / 5)
+    })
+    CATEGORICAL_EFFECTS = [-1, -1, -2, -2, 2]
+    LINEAR_TERM = np.array([
+        -0.5 + 0.01 * X['continuous'][k]
+        + CATEGORICAL_EFFECTS[X['categorical'][k]] for k in range(X.shape[0])
+    ]) + np.random.normal(0, 1, X.shape[0])
+    TRUE_PROB = expit(LINEAR_TERM)
+    Y = np.random.binomial(1, TRUE_PROB, size=N)
+    
+    return {
+        'X': X,
+        'probability_labels': TRUE_PROB,
+        'binary_labels': Y,
+        'lgb_with_binary_labels': lgb.Dataset(X, Y),
+        'lgb_with_probability_labels': lgb.Dataset(X, TRUE_PROB),
+        }"
+            );
+        }
+
+        [Test]
+        public void FbProphetTest()
+        {
+            AssetCode(
+                @"
+import pandas as pd
+from fbprophet import Prophet
+def RunTest():
+    df=pd.DataFrame({'ds': ['2007-12-10', '2007-12-11', '2007-12-12', '2007-12-13', '2007-12-14'], 'y': [9.590761, 8.519590, 8.183677, 8.072467, 7.893572]})
+    m = Prophet()
+    m.fit(df)
+    future = m.make_future_dataframe(periods=365)
+    return m.predict(future)"
+            );
+        }
+
+        [Test]
+        public void FastAiTest()
+        {
+            AssetCode(
+                @"
+from fastai.text import *
+def RunTest():
+    return 'Test is only importing the module, since available tests take too long'"
```

Comment:
Hillarious "fastai" test axed because it takes too long.

Reply:
Hopefully, we will get a use-case for algorithm trading that can be added here later.